packages/api/src/hooks/useSettings.ts

```diff
@@ -9,7 +10,8 @@ type TSetSettingsPayload = NonNullable<
 
 /** A custom hook to get and update the user settings. */
 const useSettings = () => {
-    const { data, ...rest } = useQuery('get_settings');
+    const { isSuccess } = useAuthorize();
+    const { data, ...rest } = useQuery('get_settings', { options: { enabled: isSuccess } });
```

Comment:
just to clarify:
so after being loaded 'isSuccess' will always be true, right?
what if we are doing reconnection?
may I ask you to check, if there is any issue when using this hook you try to change the language 🙏 
do we need to consider here 'isFetching' as well?

Reply:
@yauheni-deriv as per the current implementation, during language switch, there is going to be a new websocket connection. in that connection too, useSettings will be waiting for useAuthorize to be successful. i tested it, and it's working fine.
where do you suggest `isFetching` to go in the code? 🙏🏼